python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -112,26 +152,272 @@ def __init__(
         estimator: BaseEstimator,
         datasets: Dict[str, GenDataset],
         label_column: Optional[str] = None,
+        params: Optional[Dict[str, Any]] = None,
         scoring: Optional[ScoringType] = None,
         cv: Optional[CVType] = None,
+        parallelise_cv: Optional[bool] = None,
+        return_train_score_cv: bool = False,
         scaling_config: Optional[ScalingConfig] = None,
         run_config: Optional[RunConfig] = None,
         preprocessor: Optional[Preprocessor] = None,
-        resume_from_checkpoint: Optional[Checkpoint] = None,
-        **fit_params
+        **fit_params,
     ):
-        self.estimator = estimator
+        self.estimator = clone(estimator)
         self.label_column = label_column
+        self.params = params or {}
         self.fit_params = fit_params
         self.scoring = scoring
         self.cv = cv
+        self.parallelise_cv = parallelise_cv
+        self.return_train_score_cv = return_train_score_cv
         super().__init__(
             scaling_config=scaling_config,
             run_config=run_config,
             datasets=datasets,
             preprocessor=preprocessor,
-            resume_from_checkpoint=resume_from_checkpoint,
+            resume_from_checkpoint=None,
         )
 
+    def _validate_attributes(self):
+        super()._validate_attributes()
+        if TRAIN_DATASET_KEY not in self.datasets:
+            raise KeyError(
+                f"'{TRAIN_DATASET_KEY}' key must be preset in `datasets`. "
+                f"Got {list(self.datasets.keys())}"
+            )
+        if "cv" in self.datasets:
+            raise KeyError(
+                "'cv' is a reserved key. Please choose a different key "
+                "for the dataset."
+            )
+        if (
+            not isinstance(self.parallelise_cv, bool)
+            and self.parallelise_cv is not None
+        ):
+            raise ValueError(
+                "`parallelise_cv` must be a bool or None, got "
+                f"'{self.parallelise_cv}'"
+            )
```

Comment:
What about validating `estimator`, `label_column`, `params`, `scoring`, `cv` and `return_train_score_cv`?

Reply:
We don't really validate similar arguments in other trainers - there will be an exception raised by sklearn during training if either of those is incorrect.

Reply:
I think it's nice to do the validation before we kick off a distributed run, though, so we may want to do this for all trainers. I can update this when taking over